packages/shared/src/components/feeds/FeedSettings/FeedSettingsCreate.tsx

```diff
@@ -0,0 +1,256 @@
+  TypographyType,
+  TypographyTag,
+} from '../../typography/Typography';
+import { LogEvent } from '../../../lib/log';
+import { useLogContext } from '../../../contexts/LogContext';
+import { labels } from '../../../lib/labels';
+import { getFeedSettingsQueryKey } from '../../../hooks/useFeedSettings';
+import { useAuthContext } from '../../../contexts/AuthContext';
+import { PlusUser } from '../../PlusUser';
+import { ModalClose } from '../../modals/common/ModalClose';
+import { ActionType } from '../../../graphql/actions';
+
+const CreateTitle = ({ className }: { className?: string }): ReactElement => {
+  return (
+    <div className={classNames('flex items-center gap-4', className)}>
+      <Typography
+        className="tablet:typo-title3"
+        type={TypographyType.Body}
+        bold
+      >
+        New custom feed
+      </Typography>
+      <PlusUser />
+    </div>
+  );
+};
+
+const SubmitButton = (buttonProps: ButtonProps<'button'>): ReactElement => {
+  return (
+    <Button
+      {...buttonProps}
+      type="submit"
+      size={ButtonSize.Medium}
+      variant={ButtonVariant.Primary}
+    >
+      Create feed
+    </Button>
+  );
+};
+
+export const FeedSettingsCreate = (): ReactElement => {
+  const [newFeedId] = useState(() => Date.now().toString());
+  const { completeAction } = useActions();
+  const { user } = useAuthContext();
+  const router = useRouter();
+  const queryClient = useQueryClient();
+  const { displayToast } = useToastNotification();
+  const { logEvent } = useLogContext();
+  const [data, setData] = useState<CreateFeedProps>(() => ({
+    icon: '',
+  }));
+  const { createFeed } = useFeeds();
+  const { isPlus } = usePlusSubscription();
+
+  const { onFinished, delayedRedirect, isAnimating } = useProgressAnimation({
+    animationMs: 1000,
+  });
+
+  const { mutate: onSubmit, isPending: isSubmitPending } = useMutation({
+    mutationFn: createFeed,
+    onSuccess: (newFeed) => {
+      logEvent({
+        event_name: LogEvent.CreateCustomFeed,
+        target_id: newFeed.id,
+      });
+
+      queryClient.removeQueries({
+        queryKey: getFeedSettingsQueryKey(user, newFeedId),
+      });
+
+      onFinished();
+      delayedRedirect(`${webappUrl}feeds/${newFeed.id}`);
+    },
+    onError: (error) => {
+      const clientErrors = (error as ClientError)?.response?.errors || [];
+
+      if (
+        clientErrors.some(
+          (item) => item.message === labels.feed.error.feedLimit.api,
+        )
+      ) {
+        displayToast(labels.feed.error.feedLimit.client);
+
+        return;
+      }
+
+      if (
+        clientErrors.some(
+          (item) => item.message === labels.feed.error.feedNameInvalid.api,
+        )
+      ) {
+        displayToast(labels.feed.error.feedNameInvalid.api);
+
+        return;
+      }
+
+      displayToast(labels.error.generic);
+    },
+  });
+
+  useEffect(() => {
+    if (!user) {
+      return;
+    }
+
+    completeAction(ActionType.CustomFeed);
```

Comment:
Why in useEffect, don't we only want to trigger if they actually create it?

Reply:
we had this in previous /new page, it is an action that they interacted with custom feeds